code/modules/mob/living/carbon/carbon.dm

```diff
@@ -590,12 +590,18 @@
 			see_invisible = min(G.invis_view, see_invisible)
 		if(!isnull(G.lighting_alpha))
 			lighting_alpha = min(lighting_alpha, G.lighting_alpha)
-	if(dna)
+
+	var/has_xray = has_trait(TRAIT_XRAY_VISION)
+
+	if(dna && !has_xray)
 		for(var/X in dna.mutations)
 			var/datum/mutation/M = X
 			if(M.name == XRAY)

```

Comment:
might be a good moment to make the mutation give the trait directly, like hulk does